### ISSUE TITLE
IPv6 merge produces invalid results

### ISSUE THREAD

**Issue by lancehudson** [silverwind/cidr-tools#19]
testing from a repl

Minimal Example:
```
> ips = ['1001:2860:7:30e::f8/128','1001:2860:7:30e::fe/128']
[ '1001:2860:7:30e::f8/128', '1001:2860:7:30e::fe/128' ]
> merge(ips)
[
  '1001:2860:7:30e::f8/126',
  '1001:2860:7:30e::fc/127',
  '1001:2860:7:30e::fe/128'
]
```

It should return two /128 cidrs.
In case it matters node `v20.11.0`

**Comment by lancehudson** [comment]
I did notice an inconsistency in parse that may be related.

```
> parse('1001:2860:7:30e::f8/128')
[Object: null prototype] {
  single: false,
  cidr: '1001:2860:7:30e::f8/128',
  version: 6,
  prefix: '128',
  start: 21273659131713413136373978376177713400n,
  end: 21273659131713413136373978376177713400n
}
```

single should be true

**Comment by lancehudson** [comment]
Noticed that if you pass a cidr single will always be false. Summarize still errors if you send just ips and not cidrs though, so going with unrelated.

```
> parse('1001:2860:7:30e::f8');
[Object: null prototype] {
  single: true,
  cidr: '1001:2860:7:30e::f8/128',
  version: 6,
  prefix: '128',
  start: 21273659131713413136373978376177713400n,
  end: 21273659131713413136373978376177713400n
}

> merge(['1001:2860:7:30e::f8','1001:2860:7:30e::fe'])
[
  '1001:2860:7:30e::f8/126',
  '1001:2860:7:30e::fc/127',
  '1001:2860:7:30e::fe/128'
]
```

**Comment by silverwind** [comment]
> I did notice an inconsistency in parse that may be related.
> 
> ```
> > parse('1001:2860:7:30e::f8/128')
> [Object: null prototype] {
>   single: false,
>   cidr: '1001:2860:7:30e::f8/128',
>   version: 6,
>   prefix: '128',
>   start: 21273659131713413136373978376177713400n,
>   end: 21273659131713413136373978376177713400n
> }
> ```
> 
> single should be true

This is fixed, I independently noticed it today. Unforunately it does not change the output of the first case, so there's another bug likely in the merge mechanism.

**Comment by silverwind** [comment]
Sidenote: I will drop the `single` property next release, it's easy enough to do `start === end` for the same effect.

**Comment by silverwind** [comment]
Fixed in 8.0.0.
